tests/database_manager/test_database_manager.py

```diff
@@ -134,3 +167,478 @@ def test_call_queue_length_returns_queue_length(
+    def test_call_add_database_with_invalid_connection(
+        self,
+        mocked_database_constructor: MagicMock,
+        mocked_validate_connection: MagicMock,
+        database_manager: DatabaseManager,
+    ) -> None:
+        """Test add a database with an invalid connection."""
+        mocked_validate_connection.return_value = False
+        body = {
+            "id": "database_id",
+            "user": "admin",
+            "password": "12345678",
+            "host": "database_host",
+            "port": 5432,
+            "dbname": "database_name",
+            "number_workers": 8,
+        }
+        response = database_manager._call_add_database(body)
+
+        mocked_database_constructor.assert_not_called()
+        assert response == get_response(400)
+        assert "database_id" not in database_manager._databases.keys()
+
+    @patch("hyrisecockpit.database_manager.manager.Driver.validate_connection")
+    @patch("hyrisecockpit.database_manager.manager.Database")
+    def test_call_add_existing_database(
+        self,
+        mocked_database_constructor: MagicMock,
+        mocked_validate_connection: MagicMock,
+        database_manager: DatabaseManager,
+    ) -> None:
+        """Test add an existing database."""
+        mocked_validate_connection.return_value = True
+        body = {
+            "id": "database_id",
+            "user": "admin",
+            "password": "12345678",
+            "host": "database_host",
+            "port": 5432,
+            "dbname": "database_name",
+            "number_workers": 8,
+        }
+        database_manager._databases["database_id"] = "Database"  # type: ignore
+        response = database_manager._call_add_database(body)
+
+        mocked_database_constructor.assert_not_called()
+        assert response == get_response(400)
+
+    def test_call_get_databases(self, database_manager: DatabaseManager) -> None:
+        """Test get databases."""
+        database_manager._databases["db1"] = fake_database()
+        body: Dict = {}
+
+        expected_databases = [
+            {
+                "id": "db1",
+                "host": "database_host",
+                "port": 10000,
+                "number_workers": 8,
+                "dbname": "database_name",
+            }
+        ]
+        expected_response = get_response(200)
+        expected_response["body"]["databases"] = expected_databases
+        response = database_manager._call_get_databases(body)
+
+        assert response == expected_response
+
+    def test_delete_database(self, database_manager: DatabaseManager) -> None:
+        """Test delete database."""
+        database = fake_database()
+        database_manager._databases["db1"] = database
+        body: Dict = {"id": "db1"}
+        response = database_manager._call_delete_database(body)
+
+        assert get_response(200) == response
+        assert list(database_manager._databases.values()) == []
```

Comment:
I think we should check if the `_databases` dictionary has no entry under the key `db1`.

```suggestion
        assert not "db1" in database_manager._databases
```

Reply:
Probably, `not "db1" in database_manager._databases.keys()` :smirk: